subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyPluginNotationConverter.java

```diff
@@ -41,7 +42,7 @@ public void describe(DiagnosticsVisitor visitor) {
 
     @Override
     public void convert(PluginDependency notation, NotationConvertResult<? super MinimalExternalModuleDependency> result) throws TypeConversionException {
-        ModuleIdentifier module = DefaultModuleIdentifier.newId(notation.getPluginId(), notation.getPluginId() + ".gradle.plugin");
+        ModuleIdentifier module = DefaultModuleIdentifier.newId(notation.getPluginId(), PluginDependencyMarkerCoordinates.pluginName(notation.getPluginId()));
```

Comment:
I would make `PluginDependencyMarkerCoordinates.pluginName` private and have a function to create the module identifier directly so that the responsibility of the mapping is fully in `PluginDependencyMarkerCoordinates`.
```suggestion
        ModuleIdentifier module = PluginDependencyMarkerCoordinates.moduleIdentifierFor(notation);
```

Reply:
Happy to but `DefaultModuleIdentifier` is in `dependency-management`, `PluginDependencyMarkerCoordinates` in `core-api`, (how) do you do inter-project access?

Reply:
Looking at this again I think what is disturbing to me is that the knowledge that the `group` is basically the plugin identifier is spread.

Could you please add `PluginDependencyMarkerCoordinates.pluginGroup(String pluginId)` and use it?

This would move that knowledge to the same place as the rest for constructing maven coordinates for a plugin marker.